context.go

```diff
@@ -123,6 +123,7 @@ type mcontext struct {
 	idempotencyKey       string
 	username             string
 	password             string
+	headers              map[string]string
```

Comment:
I d)n't like this as header are controlled and this would allow weird shit. Please add a ClientIP property

Reply:
Plus you don't use them anywhere

Reply:
Plus yoy should use http.Header

Reply:
moved to client IP and its used now